frontend/models.py

```diff
@@ -1196,11 +1311,21 @@ def get_mol(self):
             print("Could not find molecule to update!")
 
     def save(self, *args, **kwargs):
-        old_status = self.order.status
-        old_unseen = self.order.new_status
+        if hasattr(self, "flowchart_order") and self.flowchart_order is not None:
+            old_status = self.flowchart_order.status
+            old_unseen = self.flowchart_order.new_status
+
+        elif hasattr(self, "order"):
+            old_status = self.order.status
+            old_unseen = self.order.new_status
 
         super(Calculation, self).save(*args, **kwargs)
-        self.order.update_unseen(old_status, old_unseen)
+
+        if hasattr(self, "flowchart_order") and self.flowchart_order is not None:
```

Comment:
You do not need to check if the object has these attributes, they will always be created and equal to `None` if unset. You could also verify if the `order` if is null as second case, and raise an exception if both `flowchart_order` and `order` are null.

Reply:
Updated!